docs/manual-cluster-repair.md

```diff
@@ -0,0 +1,125 @@
+# Manual cluster repair
+When one of the control-plane nodes fails, it's necessary to restore it back as
+fast as possible to avoid loosing ETCD quorum and blocking all kubeapi-server
+operations.
+
+This is a guide on how to restore you cluster back to the normal state.
+
+## One dead control-plane instance
+If you are running KubeOne cluster with machine-controller enabled, it will
+automatically recreate your failed worker nodes. But it can happen that
+cloud-instance with control-plane node fails.
+
+In this document we will cover the case when 1 (out of 3 in total) control-plane
+node is lost.
+
+### Remove dead ETCD member
+Even when one ETCD member is physically (and abruptly) removed, ETCD-ring still
+hopes it might return back online. Unfortunately this is not our case and we
+need to let ETCD-ring know that dead ETCD member is dead forever.
+
+Exec into the shell of the alive ETCD container:
+```bash
+kubectl -n kube-system exec -it ETCD-<ALIVE-HOSTNAME> sh
+```
+
+Setup client TLS authentication:
+```bash
+export ETCDCTL_API=3
+export ETCDCTL_CACERT=/etc/kubernetes/pki/ETCD/ca.crt
+export ETCDCTL_CERT=/etc/kubernetes/pki/ETCD/healthcheck-client.crt
+export ETCDCTL_KEY=/etc/kubernetes/pki/ETCD/healthcheck-client.key
+```
+
+Retrieve currently known members list (example output):
+```bash
+etcdctl member list
+2ce40012b4b4e4e6, started, ip-172-31-153-216.eu-west-3.compute.internal, https://172.31.153.216:2380, https://172.31.153.216:2379, false
+2e39cf93b81fb7ed, started, ip-172-31-153-246.eu-west-3.compute.internal, https://172.31.153.246:2380, https://172.31.153.246:2379, false
+6713c8f2e74fb553, started, ip-172-31-153-235.eu-west-3.compute.internal, https://172.31.153.235:2380, https://172.31.153.235:2379, false
+```
+
+We are going to need an ID of the dead ETCD member. To locate it, please check
+(by IP for example, or hostnames) your online instances list and find one from
+the list above, that is not online anymore.
+
+Locate and delete dead ETCD member:
+```bash
+etcdctl member remove 6713c8f2e74fb553
+Member 6713c8f2e74fb553 removed from cluster 4ec111e0dee094c3
+```
+
+Exit the ETCD pod shell.
+
+### Recreate failed instance
+Assuming you've used Terraform to provision your cloud infrastructure, use
+`terraform apply` to restore cloud infrastructure to the declared state, i.e. 3
+control-plane instances for HA clusters.
+
+From your local machine:
+```bash
+terraform apply
+```
```

Comment:
Don't you need to delete an instance manually before running `terraform apply`? As long as the cloud instance exists, `terraform apply` wouldn't do anything.